coderd/reaper/detector.go

```diff
@@ -210,11 +230,11 @@ func (d *Detector) run(t time.Time) Stats {
 	return stats
 }
 
-func unhangJob(ctx context.Context, log slog.Logger, db database.Store, pub pubsub.Pubsub, jobID uuid.UUID) error {
+func reapJob(ctx context.Context, log slog.Logger, db database.Store, pub pubsub.Pubsub, jobID uuid.UUID) error {
```

Comment:
If you wrapped the job ID in a struct like so:
```go
type reapJob struct {
  ID uuid.UUID
  JobType jobType
  Threshold time.Duration
}
```

Then you could eliminate `reapParamsFromJob`. You would convert the `database.ProvisionerJob`s to this type as soon as you read them after each query.